cunumeric/eager.py

```diff
@@ -359,6 +359,20 @@ def flip(self, rhs, axes, stacklevel):
             self.array = np.flip(rhs.array, axes)
             self.runtime.profile_callsite(stacklevel + 1, False)
 
+    def choose(self, choices, out, mode, stacklevel):
+        if self.shadow:
+            choices = self.runtime.to_eager_array(
+                choices, stacklevel=(stacklevel + 1)
+            )
+            out = self.runtime.to_eager_array(out, stacklevel=(stacklevel + 1))
+        elif self.deferred is None:
+            self.check_eager_args((stacklevel + 1), choices, out)
+        if self.deferred is not None:
+            self.deferred.choose(choices, out, stacklevel=(stacklevel + 1))
```

Comment:
"mode" is not being forwarded to the call to deferred.py:choose

Reply:
This is because I am doing mode-specific logic in [array.py](https://github.com/ipdemes/cunumeric/blob/choose_operation/cunumeric/array.py#L974-L986)
I can move it inside of the Legion task, if you think this will be better

Reply:
OK, the underlying issue here, as I commented on another line, is that the definition of `choose` on eager.py should match the definition on deferred.py. When that is the case you should be able to simply forward all arguments directly.